src/main/java/core/basesyntax/operation/BalanceHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.transaction.FruitTransaction;
+
+public class BalanceHandler implements OperationHandler {
+    private ShopServiceImpl shopServiceImpl;
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (transaction.getOperation() == FruitTransaction.Operation.BALANCE) {
+            String fruit = transaction.getFruit();
+            int quantity = transaction.getQuantity();
+            shopServiceImpl.addFruits(fruit, quantity);
+        } else {
+            throw new RuntimeException("Unsupported operation for BalanceHandler");
+        }

```

Comment:
The `apply` method should not check the operation type. The OperationStrategy should be responsible for selecting the correct handler based on the operation type. This way, the BalanceHandler will only handle BALANCE operations, and you can remove the else block which throws a RuntimeException.